packages/fxa-auth-server/scripts/convert-customers-to-stripe-automatic-tax.ts

```diff
@@ -0,0 +1,328 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import program from 'commander';
+import Stripe from 'stripe';
+import Container from 'typedi';
+
+import { setupProcessingTaskObjects } from '../lib/payments/processing-tasks-setup';
+import { AppConfig, AuthFirestore } from '../lib/types';
+
+const pckg = require('../package.json');
+
+const MONTHLY_NOTICE_DAYS = 14;
+const YEARLY_NOTICE_DAYS = 30;
+const SECONDS_TO_MILLISECONDS = 1000;
+
+// Represents a mapping of user uids to IP addresses
+interface IpAddressMap {
+  [key: string]: string;
+}
+
+// Firestore subscriptions contain additional expanded information
+// on top of the base Stripe.Subscription type
+interface FirestoreSubscription extends Stripe.Subscription {
+  customer: string;
+  plan: Stripe.Plan;
+  price: Stripe.Price;
+}
+
+const parseDryRun = (dryRun: boolean | string) => {
+  return `${dryRun}`.toLowerCase() !== 'false';
+};
+
+const parseBatchSize = (batchSize: string | number) => {
+  return parseInt(batchSize.toString(), 10);
+};
+
+const parseOutputFile = (outputFile: string) => {
+  return outputFile;
+};
+
+const isStripeTaxDisabled = (subscription: Stripe.Subscription) => {
+  return !subscription.automatic_tax.enabled;
+};
+
+// Checks whether subscription renews too soon to be upgraded to
+// Stripe Automatic Tax
+// 1 month plans require at least 14 days of notice
+// 6 month and yearly plans require at least 30 days of notice
+// This function does not support plans with intervals of less than 1 month
+const isWithinNoticePeriod = (sub: Stripe.Subscription) => {
+  const noSoonerThan = new Date();
+  if (sub.items.data[0].plan.interval === 'month') {
```

Comment:
At the risk of sounding ignorant (versus just stupid), what are the interval options? Like, https://stripe.com/docs/api/plans/object#plan_object-interval ?
Do we need to potentially consider a daily or weekly subscription and use a `switch/case` instead?

Reply:
>  Do we need to potentially consider a daily or weekly subscription and use a switch/case instead?
I explicitly state before the function that it does not support intervals smaller than 1 month.

I haven't heard from legal what the notice period would need to be on intervals smaller than 1 month. I'm also not sure that we have any plans with an interval of less than 1 month.